gt/main.c

```diff
@@ -550,18 +673,52 @@ gt_proc(void *arg)
 				continue;
 			}
 
+			/*
+			 * If it is a fragmented packet,
+			 * then try to reassemble.
+			 */
+			if (pkt_info.frag == true) {
+				complete_pkt = gt_reassemble_incoming_pkt(
+					m, cur_tsc, &pkt_info, instance);
+
+				/* Process the death packets. */
+				process_death_row(socket, port,
+					tx_queue, instance, gt_conf);
+
+				if (complete_pkt == NULL)
+					continue;
+
+				ret = gt_parse_incoming_pkt(
+					complete_pkt, &pkt_info);
+				if (ret < 0) {
+					if (pkt_info.outer_ip_ver ==
+							ETHER_TYPE_IPv6) {
+						add_pkt_ipv6_acl(
+							&acl, complete_pkt);
+						continue;
+					}
+
+					RTE_LOG(ALERT, GATEKEEPER,
+						"gt: parsing an invalid packet!\n");
+					rte_pktmbuf_free(complete_pkt);
```

Comment:
Since this check to add to the ACL is the same as above, should this code be made into an inline function so we can avoid duplicating this code?